server/commands/list_objects.go

```diff
@@ -260,25 +260,29 @@ func (q *ListObjectsQuery) performChecks(
 	var objectsFound = new(uint32)
 
 	// iterate over all object IDs in the store and check if the user has relation with each
+ForLoop:
 	for {
-		object, err := iter.Next()
-		if err != nil {
-			if errors.Is(err, storage.ErrIteratorDone) {
-				break
-			} else {
-				errChan <- err
-				return
+		select {
+		case <-ctx.Done():
+			break ForLoop
```

Comment:
a simple `break` would break out of the `select` but not the `for` loop. And a `return` would exit the function :(

Reply:
No, the channel should only be closed in the specific scenario where we are sure we sent all the `object ids` to it.